controlplane/src/core/repositories/GraphCompositionRepository.ts

```diff
@@ -46,21 +48,125 @@ export class GraphCompositionRepository {
+      composition.targetId,
+      composition.schemaVersionId,
+    );
+
+    return {
+      id: composition.id,
+      schemaVersionId: composition.schemaVersionId,
+      createdAt: composition.createdAt.toISOString(),
+      isComposable: composition.isComposable || false,
+      compositionErrors: composition.compositionErrors || undefined,
+      createdBy: composition.createdBy || undefined,
+      isCurrentDeployed,
+    };
+  }
+
+  public async getCompositionSubgraphs(input: { compositionId: string }) {
+    const compositionSubgraphs = await this.db
+      .select({
+        id: graphCompositionSubgraphs.id,
+        schemaVersionId: graphCompositionSubgraphs.schemaVersionId,
+        name: targets.name,
+      })
+      .from(graphCompositionSubgraphs)
+      .innerJoin(graphCompositions, eq(graphCompositions.id, graphCompositionSubgraphs.graphCompositionId))
+      .innerJoin(schemaVersion, eq(schemaVersion.id, graphCompositionSubgraphs.schemaVersionId))
+      .innerJoin(targets, eq(targets.id, schemaVersion.targetId))
+      .where(eq(graphCompositions.id, input.compositionId))
+      .execute();
+
+    return compositionSubgraphs;
+  }
+
+  public async getGraphCompositions({
+    fedGraphTargetId,
+    organizationId,
+    limit,
+    offset,
+    startDate,
+    endDate,
+  }: {
+    fedGraphTargetId: string;
+    organizationId: string;
+    limit: number;
+    offset: number;
+    startDate: string;
+    endDate: string;
+  }): Promise<GraphCompositionDTO[]> {
+    const fedRepo = new FederatedGraphRepository(this.db, organizationId);
+
+    const resp = await this.db
+      .select({
+        id: graphCompositions.id,
+        schemaVersionId: graphCompositions.schemaVersionId,
+        isComposable: graphCompositions.isComposable,
+        compositionErrors: graphCompositions.compositionErrors,
+        createdAt: graphCompositions.createdAt,
+        createdBy: users.email,
+      })
+      .from(graphCompositions)
+      .innerJoin(schemaVersion, eq(schemaVersion.id, graphCompositions.schemaVersionId))
+      .leftJoin(users, eq(graphCompositions.createdBy, users.id))
+      .where(
+        and(
+          eq(schemaVersion.targetId, fedGraphTargetId),
+          gt(graphCompositions.createdAt, new Date(startDate)),
+          lt(graphCompositions.createdAt, new Date(endDate)),
+        ),
+      )
+      .orderBy(desc(schemaVersion.createdAt))
+      .limit(limit)
+      .offset(offset)
+      .execute();
+
+    const compositions: GraphCompositionDTO[] = [];
+
+    for (const r of resp) {
+      const isCurrentDeployed = await fedRepo.isLatestValidSchemaVersion(fedGraphTargetId, r.schemaVersionId);
+
+      compositions.push({
+        id: r.id,
+        schemaVersionId: r.schemaVersionId,
+        createdAt: r.createdAt.toISOString(),
+        isComposable: r.isComposable || false,
+        compositionErrors: r.compositionErrors || undefined,
+        createdBy: r.createdBy || undefined,
+        isCurrentDeployed,
```

Comment:
We don't know if it was deployed. We only know it is latest.

Reply:
do you mean to rename it to latestValid?

Reply:
Done